sydtest/src/Test/Syd/Def/SetupFunc.hs

```diff
@@ -91,6 +91,12 @@ setupAroundWith' f = aroundWith' $ \takeBoth outer oldInner ->
   let SetupFunc provideNewInner = f outer oldInner
    in provideNewInner $ \newInner -> takeBoth outer newInner
 
+-- | Use all outers and inner to provide a new inner
+-- This is a more constrained version of 'setupAroundWith'' to more easily allow providing an inner based on multiple outers
+-- To get a specific outer you can use 'getElem'
```

Comment:
Aha, I'll have to look at patternmatching the HList then, as I used the `getElem` myself here. First time I come in contact with a type level list I think, at least this closely :sweat_smile:

Reply:
And about the example, yea it might indeed, I feel like people will just find this when the bump into the same problem I had, and at that point they have their own example :P For now I just removed this line.